ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java

```diff
@@ -367,6 +379,178 @@ public Map apply(@Nullable String fqdn) {
+                    subresult.put((String) ((Map.Entry) subconfig).getKey(), ((Map.Entry) subconfig).getValue());
+                }
+                result.put((String) config.getKey(), subresult);
+            } else {
+                Object value = config.getValue() instanceof Map ? ((Map) config.getValue()).keySet().toArray()[0] : config.getValue();
+                result.put((String) config.getKey(), value);
+            }
+        }
+
+        return result;
+    }
+
+    public Map<String, Integer> listAlertTargets() {
+        AlertTargets alertTargets = restAdapter.create(AlertTargetEndpoint.class).listAlertNotifications();
+
+        Map<String, Integer> result = MutableMap.of();
+        for (Map<String, Object> item: alertTargets.getItems()) {
+            result.put((String) ((Map) item.get("AlertTarget")).get("name"), ((Double) ((Map) item.get("AlertTarget")).get("id")).intValue());
+        }
+
+        return result;
+    }
+
+    public void updateAlertTarget(Integer targetId, Map<String, Object> ambariAlertNotifications) {
+        ImmutableMap<String, Map<String, Object>> alertTargetRequest = ImmutableMap.of("AlertTarget", ambariAlertNotifications);
+
+        restAdapter.create(AlertTargetEndpoint.class).editAlertNotification(targetId, alertTargetRequest);
+    }
+
+    private Map<String, Object> getAmbariAlertsConfigFromProperties() {
+        ConfigBag args = new ConfigBag();
+        Map<String, Object> properties = MutableMap.of();
+        for (Map.Entry<ConfigKey<?>, Object> entry: getManagementContext().getConfig().getAllConfig().entrySet()) {
+            String key = entry.getKey().getName();
+            if (key.startsWith(AmbariCluster.AMBARI_ALERTS_CONFIG_PREFIX)) {
+                key = Strings.removeAllFromStart(key,AmbariCluster.AMBARI_ALERTS_CONFIG_PREFIX);
+
+                if (key.equals(AmbariCluster.AMBARI_ALERTS_NOTIFICATION_ALERT_STATES.getName())) {
+                    args.putStringKey(key,toArray((String) entry.getValue()));
+                } else if (key.equals(AmbariCluster.AMBARI_ALERTS_NOTIFICATION_RECIPIENTS.getName())){
+                    args.putStringKey(key,formatAmbariAlertsNotificationRecipients((String) entry.getValue()));
+                }else if (key.startsWith(AmbariCluster.AMBARI_ALERTS_NOTIFICATION_PROPERTIES_PREFIX)) {
+                    key = Strings.removeAllFromStart(key, AmbariCluster.AMBARI_ALERTS_NOTIFICATION_PROPERTIES_PREFIX);
+                    properties.put(key, entry.getValue());
+                } else {
+                    args.putStringKey(key, entry.getValue());
+                }
+            }
+        }
+
+        if (!properties.isEmpty()) {
+            args.putStringKey(Strings.removeAllFromEnd(AmbariCluster.AMBARI_ALERTS_NOTIFICATION_PROPERTIES_PREFIX, "."), properties);
+        }
+
+        return args.getAllConfig();
+    }
+
+    private String[] toArray(String input) {
+        String value = input.replaceAll("[\\[\\]\\s]+", "");
```

Comment:
Do not implement such functions yourself.
Use GSON parser or some other meaningful utility.